src/ansys/fluent/core/codegen/allapigen.py

```diff
@@ -45,12 +45,15 @@ def generate(version: str, static_infos: dict, verbose: bool = False):
     """Generate Fluent API classes."""
     api_tree = {"<meshing_session>": {}, "<solver_session>": {}}
     _update_first_level(api_tree, tuigen.generate(version, static_infos, verbose))
+    api_tree = {
+        k: dict(v) for k, v in api_tree.items()
+    }  # Ensure all values are dicts for JSON serialization
     _update_first_level(api_tree, datamodelgen.generate(version, static_infos, verbose))
     _update_first_level(api_tree, settingsgen.generate(version, static_infos, verbose))
     api_tree_file = get_api_tree_file_name(version)
     Path(api_tree_file).parent.mkdir(parents=True, exist_ok=True)
-    with open(api_tree_file, "wb") as f:
-        pickle.dump(api_tree, f)
+    with open(api_tree_file, "w") as f:
+        json.dump(api_tree, f)
```

Comment:
I think we were no longer writing the pickle file. Some code can be cleaned up here (in a future PR).

Reply:
Sure.

Reply:
https://github.com/ansys/pyfluent/issues/3739